### PR TITLE
Add slack alerts on job failure

### PR DESCRIPTION
**Pull request by jcox-dev**
# What has changed

- Added slack alerts on most job failures
- Added slack alerts on terraform start/success/fail
- Expects new `((slack.webhook))` config (TODO: add this)

# How to test?

Carefully read **each and every** line that has been added (of which there are 2,815) to make sure everything is as it seems.

OR

`fly validate-pipeline -c pipelines/*.yml`

### CODE REVIEW

**Review by kierenbeckett — COMMENTED** [review]
:+1: Let me know if you need `slack.webhook` adding to the central concourse

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
Could save a few code lines by defining the slack on failure component with an alias so it can be reused, like:
```yaml
slack_failure_alert : &slack_failure_alert 
      put: slack-alert
      params:
        icon_emoji: ":concourse:"
        username: Concourse
        attachments: [
              {
                  "fallback": "$BUILD_JOB_NAME failed. See build: $ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME",
                  "title": "$BUILD_JOB_NAME failed",
                  "title_link": "$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME",
                  "fields": [
                      {
                          "title": "Pipeline",
                          "value": "$BUILD_PIPELINE_NAME",
                          "short": true
                      },
                      {
                          "title": "Build",
                          "value": "#$BUILD_NAME",
                          "short": true
                      }
                  ],
                  "color": "#F35A00"
              }
          ]
```

Then referred to like
```yaml
on_failure: *slack_failure_alert
```

**Review by AdamHawtin — APPROVED** [review]
LGTM 👍

**Review by ryangrundy7 — APPROVED** [review]
LGTM 👍

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+sudo: required
+
+install:
+  - wget https://github.com/concourse/concourse/releases/download/v5.4.0/fly-5.4.0-linux-amd64.tgz
+  - tar -xvzf fly-5.4.0-linux-amd64.tgz
+
+script:
+  - for i in pipelines/*.yml; do ./fly validate-pipeline -c $i; done
```
Comment (AdamHawtin) [inline .travis.yml:8]: 👍

**Pull request merged**